graphql_compiler/schema_merging/utils.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql.language.visitor import Visitor, visit
+
+
+class SchemaError(Exception):
+    pass
+
+
+SchemaData = namedtuple('SchemaData', ['query_type', 'scalars', 'directives', 'has_extension'])
+
+
+class GetSchemaDataVisitor(Visitor):
```

Comment:
All this information, but the `has_extension` part is easily available in the `GraphQLSchema` so I am not sure this class is necessary.

Reply:
That's a good point, I'll use the information in GraphQLSchema for most of that. For extensions, I'll no longer check for them beforehand and raise an error if I run into one while traversing the ast.

Reply:
One downside of doing this though is that I need to go through transforming the ast into a GraphQLSchema, instead of directly working with the ast. It's not too much of a downside right here since I do build_ast_schema earlier anyway, but later in the merging step, I use the visitor to get schema data a bunch without ever turning the ast into a GraphQLSchema. So it might be useful to keep this.

Reply:
Once we've already called `build_ast_schema()` for its validation logic, we might as well keep the result and use it to simplify the rest of our code. I also think your idea of simply raising errors on extension types if/when they are encountered is a good one.

Overall, the simpler and more understandable we can make the code, the easier it will be to test and maintain in the long run. This is code that will be constantly called, in production, for many years to come. So let's do our future selves a favor 😃

Reply:
Ok then either option makes sense to me.